routemaster/jobs/batch.rb

```diff
@@ -0,0 +1,57 @@
+require 'routemaster/jobs'
+require 'routemaster/models/batch'
+require 'routemaster/services/deliver'
+require 'routemaster/mixins/log'
+
+module Routemaster
+  module Jobs
+    class Batch
+      include Mixins::Log
+
+      def initialize(delivery: Services::Deliver)
+        @delivery = delivery
+      end
+
+      def call(uid)
+        batch = Models::Batch.new(uid: uid)
+
+        # prevents further ingestion in this batch (idempotent)
+        batch.promote
+
+        subscriber = batch.subscriber
+        data = batch.data
+
+        # handle unsubscription, autodrop
+        if subscriber.nil? || data.nil? || data.empty?
+          batch.delete
+          return self
+        end
+
+        events = data.
+          map { |d| Services::Codec.new.load(d) }.
+          select { |msg| msg.kind_of?(Models::Event) }
```

Comment:
Not sure I 100% agree here. At the moment, `Batch` is completely ignorant of the structure of messages (they're just raw blobs). The data structure of the batch itself is abstracted (which Redis keys, how they're indexed); but the structure of the _payloads_ is (currently, from the perspective of `Batch`) irrelevant.

I'm not inclined on adding `Event` serialisation/deserialisation into `Batch` itself as a) it's not necessary for `Batch` to work and b) it's already quite meaty.
Do you have an alternate suggestion? Slapping an abstraction layer atop `Batch`, maybe subclassing into `EventBatch`? Something else?